src/icrar/leap-accelerate/MetaData.cc

```diff
@@ -86,32 +86,31 @@ namespace icrar
         }
 
         Vector<double> range(2, 0.0);
-        if(msc.observation().nrow() > 0)
+        if(msc->observation().nrow() > 0)
         {
-            msc.observation().timeRange().get(0, range);
+            msc->observation().timeRange().get(0, range);
         }
         //start_time = range[0];
         //end_time = range[1];
 
 
-        casacore::Vector<double> time = msmc.time().getColumn();
+        casacore::Vector<double> time = msmc->time().getColumn();
         //msmc.time();
 
         this->nantennas = 4853; //TODO
-        casacore::Vector<std::int32_t> a1 = msmc.antenna1().getColumn()(Slice(0, 4853, 1)); //TODO
-        casacore::Vector<std::int32_t> a2 = msmc.antenna2().getColumn()(Slice(0, 4853, 1)); //TODO
+        casacore::Vector<std::int32_t> a1 = msmc->antenna1().getColumn()(Slice(0, 4853, 1)); //TODO
+        casacore::Vector<std::int32_t> a2 = msmc->antenna2().getColumn()(Slice(0, 4853, 1)); //TODO
 
         //Start calculations
-
-        casacore::Matrix<double> A;
-        casacore::Array<std::int32_t> I;
         casacore::Matrix<double> A1;
         casacore::Array<std::int32_t> I1;
-        std::tie(A, I) = icrar::cpu::PhaseMatrixFunction(a1, a2, 0);
-        casacore::Matrix<double> Ad = icrar::cpu::RightInvert(A);
+        std::tie(A1, I1) = icrar::cpu::PhaseMatrixFunction(a1, a2, 0);
```

Comment:
`auto [A1, I1] = icrar::cpu::PhaseMatrixFunction(a1, a2, 0);` should do if you are using 17.

The top-level `CMakeLists.txt` has `set(CMAKE_CXX_STANDARD 17)`, but then the inner `CMakeLists.txt` under `src/icrar/leap-accelerate` has ` target_compile_features(${TARGET_NAME} PUBLIC cxx_std_11)`. Which one is it?

Reply:
I'm surprised g++ 6 didn't complain about the 17 standard for the other targets. As far as I've checked the cluster runs g++ 6 which should fully support up to c++14 according to this: https://gcc.gnu.org/projects/cxx-status.html#cxx2a

I'll try updating all to C++14 test a basic feature from it such as [[deprecated]]